packages/cli/src/cmds/compare/ChangeReporter.ts

```diff
@@ -222,23 +258,91 @@ export default class ChangeReporter {
     assert(this.appMapMetadata);
     const failedAppMaps = new Set<AppMapName>();
     {
-      for (const appmap of await this.appmapData.appmaps(RevisionName.Head)) {
+      for (const appmap of await this.paths.appmaps(RevisionName.Head)) {
         const metadata = this.appMapMetadata[RevisionName.Head].get(appmap);
         if (!metadata) {
           console.log(prominentStyle(`Metadata for ${appmap} not found!`));
           continue;
         }
-        if (metadata.test_status === 'failed') {
-          this.referencedAppMaps.add(RevisionName.Head, appmap);
-          failedAppMaps.add(appmap);
-        }
+        if (metadata.test_status === 'failed') failedAppMaps.add(appmap);
       }
     }
     this.failedAppMaps = failedAppMaps;
   }
 }
 
-class ReportGenerator {
+// Gets a function that returns true if the given appmap is a test.
+function isTest(appMapMetadata: AppMapMetadata): (appmap: AppMapName) => boolean {
+  return (appmap: AppMapName): boolean => {
+    const metadata = appMapMetadata[RevisionName.Head].get(appmap);
+    return !!(metadata && metadata.recorder.type === 'tests');
+  };
+}
+
+// Selects AppMaps that have stable names, and are in the head revision but not in the base revision.
+export function isNew(
+  baseAppMaps: Set<AppMapName>,
+  isTestFn: (appmap: AppMapName) => boolean
+): (appmap: AppMapName) => boolean {
+  return (appmap: AppMapName) => isTestFn(appmap) && !baseAppMaps.has(appmap);
+}
+
+// Selects AppMaps that are present in both the base and head revisions, and have different digests.
+export function isChanged(
+  baseAppMaps: Set<AppMapName>,
+  isTestFn: (appmap: AppMapName) => boolean,
+  digests: Digests
```

Comment:
This is actually a fix also - `isChanged` wasn't checking the digests before. Oops.